### ISSUE TITLE
VOC_BASE_CNT IN faster_rcnn.py

### ISSUE THREAD

**Issue by Zhengzkang** [Phoenix-V/DiGeo#1]
Hi bro, I would be very grateful if you could tell me what VOC_BASE_CNT stands for?  which value is set by what？

**Comment by Phoenix-V** [comment]
Thank you for your interest. I believe you are referring to https://github.com/Phoenix-V/DiGeo/blob/master/fsdet/modeling/roi_heads/fast_rcnn.py#L46.

These are the number of instances for each base class in VOC. The numbers can be automatically obtained according to the output by the dataloader when you run the code. I just pre-store them for the convenience of usage.

**Comment by Zhengzkang** [comment]
帅哥，你好，看到你们团队发了许多顶会，如fct，meta faster rcnn，十分佩服。此外，如果我总共有9个类别，3个做新类的自建数据集，规模和VOC差不多，是不是需要把这个5改成3？这行代码在fast_rcnn.py中222行
cls_count = VOC_BASE_CNT[split] + [n_shot for _ in range(3)] + [self.cfg.LOSS.ADJUST_BACK,]
但是我发现自蒸馏的结果尤其是新类的，还是不太理想；还有就是这个训练时间学习时间太长了，效率方面的确不尽人意，希望大佬团队能够往效率方面研究，以供我们学习。

**Comment by Phoenix-V** [comment]
Thank you for your interest in our team. Yes, you should change 5 to 3. Basically, the cls_count is used to store the number of instances per class, for both base and novel. The value in cls_count should also follow the order of classes that you predefined.

**Comment by Phoenix-V** [comment]
One note about hyper-parameter adjustment is the REPEAT_THRESHOLD and ADJUST_BACK. Having them set properly is important. Regarding efficiency, thank you for your suggestion. You can actually do a pre-training on base first and then do joint training on all classes. However, please make sure to derive the ETF classifier for all classes based on the classifier you obtained during base pre-training. As an alternative, you can directly ETF as the classifier even in the base-training stage.

**Comment by Zhengzkang** [comment]
十分感谢你的回复，我的代码能力太差劲了，我尝试了很多方法，但是都未能在Detectron2中实现混淆矩阵的绘制，大佬能提供一下代码和具体步骤？再一次感谢你！苦比研二硕士，被导师push做FSOD，还准备让我发顶刊，太难了

**Comment by Phoenix-V** [comment]
你说的是confusion matrix？我没画过那个东东诶。。如果是说Fig. 3里面的那个图的话，那个就是classifier weights之间dot-product一下，然后直接plt.imshow()就可以了的。我也不会在detectron2里面直接做任何的操作，但是你可以尝试把所有的prediction/model-parameters给save下来，然后再额外写一个code去处理这些prediction/parameters
